src/pynxtools/nomad/parser.py

```diff
@@ -487,8 +488,10 @@ def parse(
             if getattr(archive.nexus, var, None) is not None:
                 app_def = var
                 break
-        archive.metadata.entry_type = app_def
-        archive.metadata.domain = "nexus"
+        # If NeXus subsection written under the other schema
+        if not extend_archive and not archive.metadata.entry_type:
```

Comment:
Do we need the need argument here? I guess if any other schema sets the `entry_type` by itself, then simply checking for `if not archive.metadata.entry_type` should be sufficient, no? Not sure if there is any case where that's not applicable.

Reply:
I also thought I could check if `entry_type` was filled or not. 

But, I added the argument so that it does not break any in any cases which may be out of my knowledge.

Reply:
I suggest (in the interest of code maintainability) to **not** add the `extend_archive` parameter and instead just try if the  `not archive.metadata.entry_type` is sufficient. We can always add the parameter in case there is an actual use case for it, but I don't like having it for the sake that it might be needed.